Makefile

```diff
@@ -1,83 +1,38 @@
-
-BUILD_TAG ?= notifications-nodejs-client-manual
-
-DOCKER_CONTAINER_PREFIX = ${USER}-${BUILD_TAG}
-
-.PHONY: help
 help:
 	@cat $(MAKEFILE_LIST) | grep -E '^[a-zA-Z_-]+:.*?## .*$$' | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-.PHONY: dependencies
-dependencies:  ## Install build dependencies
-	npm install
-
-.PHONY: build
-build: dependencies ## Build project
-
-.PHONY: test
-test: ## Run tests
-	npm test
+ci:
+	$(NPM) ci --production
 
-.PHONY: integration-test
-integration-test: ## Run integration tests
-	npm test --integration
+ci-dev:
+	$(NPM) ci
 
-.PHONY: generate-env-file
-generate-env-file: ## Generate the environment file for running the tests inside a Docker container
-	script/generate_docker_env.sh
+install:
+	$(NPM) install --production
 
-.PHONY: prepare-docker-runner-image
-prepare-docker-runner-image: ## Prepare the Docker builder image
-	make -C docker build
+install-dev:
+	$(NPM) install
 
-.PHONY: build-with-docker
-build-with-docker: prepare-docker-runner-image ## Build inside a Docker container
-	docker run -i --rm \
-		--name "${DOCKER_CONTAINER_PREFIX}-build" \
-		-v "`pwd`:/var/project" \
-		-e http_proxy="${HTTP_PROXY}" \
-		-e HTTP_PROXY="${HTTP_PROXY}" \
-		-e https_proxy="${HTTPS_PROXY}" \
-		-e HTTPS_PROXY="${HTTPS_PROXY}" \
-		-e NO_PROXY="${NO_PROXY}" \
-		${DOCKER_BUILDER_IMAGE_NAME} \
-		make build
+build:
+	$(NPM) run build ## Build project
 
-.PHONY: test-with-docker
-test-with-docker: prepare-docker-runner-image generate-env-file ## Run tests inside a Docker container
-	docker run -i --rm \
-		--name "${DOCKER_CONTAINER_PREFIX}-test" \
-		-v "`pwd`:/var/project" \
-		-e http_proxy="${HTTP_PROXY}" \
-		-e HTTP_PROXY="${HTTP_PROXY}" \
-		-e https_proxy="${HTTPS_PROXY}" \
-		-e HTTPS_PROXY="${HTTPS_PROXY}" \
-		-e NO_PROXY="${NO_PROXY}" \
-		--env-file docker.env \
-		${DOCKER_BUILDER_IMAGE_NAME} \
-		make test
+test: ## Run tests
+	$(NPM) test
 
-.PHONY: integration-test-with-docker
-integration-test-with-docker: prepare-docker-runner-image generate-env-file ## Run integration tests inside a Docker container
-	docker run -i --rm \
-		--name "${DOCKER_CONTAINER_PREFIX}-integration-test" \
-		-v "`pwd`:/var/project" \
-		-e http_proxy="${HTTP_PROXY}" \
-		-e HTTP_PROXY="${HTTP_PROXY}" \
-		-e https_proxy="${HTTPS_PROXY}" \
-		-e HTTPS_PROXY="${HTTPS_PROXY}" \
-		-e NO_PROXY="${NO_PROXY}" \
-		--env-file docker.env \
-		${DOCKER_BUILDER_IMAGE_NAME} \
-		make integration-test
+integration-test: ## Run integration tests
+	$(NPM) test --integration
 
-.PHONY: clean-docker-containers
-clean-docker-containers: ## Clean up any remaining docker containers
-	docker rm -f $(shell docker ps -q -f "name=${DOCKER_CONTAINER_PREFIX}") 2> /dev/null || true
+publish:
+	$(NPM) publish --access public
 
 clean:
-	rm -rf .cache venv
+	$(NPM) run clean
+
+.PHONY:
+	install install-dev build \
+	test integration-test publish-to-npmjs \
+	ci ci-dev clean
```

Comment:
Add targets for `fmt` and use prettier.

Reply:
I'm planning to run pretty/fmt in a separate PR that contains only format/pretty changes.